treestartr blog post

wrightaprilm opened the pull request:
Hi all (and ping @stefaniebutland),

Here's a draft of my treestartr post. I might make a few more tweaks (like getting new lab members on to my site and actually linking to their bios in the piece).  Let me know if this is roughly what you're looking for.

stefaniebutland commented:
Thank you for submitting this @wrightaprilm. Quick scan gave me a very big smile. I'll do a proper review by end of day Tues Dec 4.

stefaniebutland commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,101 @@
+I'm presently working on a dated phylogeny with ants. Two of my lab members, Christina Kolbmann and Tyler Tran, are interested in fossil taxon sampling and its relationship to divergence time estimation accuracy and precision. Ants have tons of fossils, including multiple fossils in some genera or subfamilies. This is great - it means we have lots of data. But not everyone else does. This wealth  ...
+
+In these cases, we know roughly where on the tree might be a reasonable starting place to try attaching the taxon. In this case, we can use the function [`text_placr`](https://ropensci.github.io/treeStartR/#adding-tips-via-csv) to read in a dataframe of taxonomic relationships. We can then use those relationships to generate a starting tree. An example taxonomy dataframe is below. In this data fr ...
+
```
Any chance you could add a figure/ sketch of the tree you are describing in the text so that readers can picture "tree", "tips", "monophyletic group"? I'm familiar with the terms but still had trouble picturing it from the data frame and text without getting distracted from the flow of the post.

stefaniebutland commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,101 @@
+
+For some of our workflows in the lab, we use multiple functions. For example, we might first place all the tips that have close relatives using `present_tippr`. Then, tips that represent more distantly related taxa can be placed via `text_placr`, and the leftover fossils added via `rand_absent_tippr`. 
+
+## Why ROpenSci
```
ROpenSci -> rOpenSci 
throughout

stefaniebutland commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,101 @@
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+
+And I hope I can answer this question adequately. I picked ROpenSci because it was explicitly R (so the editor and reviewers would know best practices) and the guidelines and how review would proceed were very clear. And for a totally new experience of in-depth code review, being able to see where the effort was collaborative was helpful. I looked at [other journals](https://joss.theoj.org/about# ...
```
Note the content at the JOSS link (https://joss.theoj.org/about#author_guidelines) has a new location https://joss.readthedocs.io/en/latest/submitting.html

stefaniebutland replied:
" I chose Dave & Rachel ..." Could you say "suggested"?

stefaniebutland commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,101 @@
+
+And I hope I can answer this question adequately. I picked ROpenSci because it was explicitly R (so the editor and reviewers would know best practices) and the guidelines and how review would proceed were very clear. And for a totally new experience of in-depth code review, being able to see where the effort was collaborative was helpful. I looked at [other journals](https://joss.theoj.org/about# ...
+
+ROpenSci also has a collaboration with [Methods in Ecology and Evolution](https://besjournals.onlinelibrary.wiley.com/hub/journal/2041210X/author-guidelines). I plan to send a short software note there. It's very nice to have the flexibility to send the paper to a traditional outlet like MEE, or to send it to [JOSS](https://joss.theoj.org/), or to do nothing at all, if I didn't want to. 
```
maybe link `[collaboration](https://ropensci.org/blog/2017/11/29/review-collaboration-mee/)`

please spell MEE the second time, rather than abbreviation

stefaniebutland commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,101 @@
+
+## Future
+
+I'll likely add in more utility functions for parsing biological data for use with divergence time estimation. Particularly, I often work with [RevBayes](https://revbayes.github.io/), which is a sandbox-like framework for Bayesian analysis of phylogeny and comparative methods. Our analyses integrate across fossil taxa from databases, molecular data from the literature and stratigraphic ranges. Ea ...
```
Can you change the order of these phrases? It sounds like you want pull requests to your contributing guide
" I have a Contributing guide here; please feel free to make a pull request or file an issue if there’s something you’d like to see added."

wrightaprilm commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,101 @@
+date: 2018-12-11
+categories: blog
+topicid: TODO
+tags:
```
```suggestion
tags:
- software-peer-review
```

wrightaprilm commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,102 @@
+package_version: 0.1.0
+authors:
+  - name: April Wright
+    url: https://ropensci.github.io/treeStartR/
```
```suggestion
    url: https://paleantology.com/the-wright-lab/
```

wrightaprilm commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,102 @@
+package_version: 0.1.0
+authors:
+  - name: April Wright
+    url: https://paleantology.com/the-wright-lab/
```
```suggestion
    url: https://paleantology.com/the-wright-lab/
    twitter: WrightingApril
```

wrightaprilm commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,101 @@
+date: 2018-12-11
+categories: blog
+topicid: TODO
+tags:
```
Ooooh, good catch.

stefaniebutland commented:
@wrightaprilm Please let me know when you're finished addressing comments - any time up to Mon Dec 10 - and I'll have a last look before publishing.

wrightaprilm commented:
I think I've addressed all your comments. Go ahead and look it over - I also sent it to the reviewers to have a look, since they're mentioned quite a bit towards the end.

wrightaprilm commented:
OK, heard back from all the package reviewers and made changes as suggested by them. I'm happy if you're happy!

stefaniebutland commented:
> heard back from all the package reviewers and made changes as suggested by them

This is great!!

I think a few of my suggestions are in unresolved "hidden conversations". I'm sorry about that - it's possible that my way of adding comments isn't clear enough to the person receiving the review. Copying them here.

![screen shot 2018-12-07 at 2 55 35 pm](https://user-images.githubusercontent.com/11927811/49676823-3aff8a80-fa30-11e8-9b28-b5d22fe046fc.png)

- "Ants have tons of fossils" Can you rephrase to indicate there are lots of fossilized ants?
I don't want to alter your writing style so you can choose to ignore this.
- Can you add a line to say what you do once you have a starter tree? I (somewhat naive reader) felt like that was missing in the emphasis on getting a starter tree
- "But not everyone else does." Can you omit this line? It's a fact, but in reading it, I felt like something else related to it was coming up in the text
- Any chance you could add a figure/ sketch of the tree you are describing in the text so that readers can picture "tree", "tips", "monophyletic group"? I'm familiar with the terms but still had trouble picturing it from the data frame and text without getting distracted from the flow of the post.

This last request might be a bit much to request at this late stage (Friday!) if you don't already have something ready, so it's a wishlist item for clarity but I won't insist on it! 

It's a really nicely written post so it's 100% fine to say "let's publish without further changes".

wrightaprilm commented:
Oh, no problem, I actually have a great figure for there being tons of ants. Always excited to use it. 

I'm a little old school with github, and sometimes miss multiple comments. Sometimes I feel a little bad for my Data Carpentry dc-ecology-py comaintainer ;P

stefaniebutland commented:
> have a great figure for there being tons of ants.

key issue is for non-phylo familiar folk to see what the terms mean and therefore be able to picture where the placements in your table would go

wrightaprilm commented:
I ended up not adding the ant figure, but contextualizing what it means to have lots of fossils, and why that's important for my personal research. Also added a phylogeny figure.

stefaniebutland commented on content/blog/2018-12-11-treestartr.md:
```diff
@@ -0,0 +1,104 @@
+
+I'm presently working on a dated phylogeny of ants. Two of my [lab members](https://paleantology.com/the-wright-lab/), Christina Kolbmann and Tyler Tran, are interested in fossil taxon sampling and its relationship to divergence time estimation accuracy and precision. Ants have tons of fossils, including multiple fossils in some genera or subfamilies. Many of them are preserved in amber, leading  ...
+
+![](../../themes/ropensci/static/img/blog-images/2018-121-11-treestartr/tree.png)
```
```suggestion
![](/img/blog-images/2018-12-11-treestartr/tree.png)
```

stefaniebutland commented:
@wrightaprilm the added ant fossils text is great.

I made a "suggested change" in your pull request to fix typo in path to phylo fig (date part of folder name) that prevented it from being viewed, and also changed the way the path is listed. The "suggested change" is a new GitHub feature that, if you accept, is automatically committed (only the 2nd time I've used this). fyi suggested path is `![](/img/blog-images/2018-12-11-treestartr/tree.png)`.

You can check the final product in preview: https://deploy-preview-353--ropensci.netlify.com/blog/2018/12/11/treestartr/

If for some reason the image doesn't look right (size etc), I can help, and maybe can make the changes myself to save you the back and forth.

Cheers!

wrightaprilm commented:
I love the suggestion mechanism!

It would have been such a clean fix ... if the actual name of the directory didn't contain a typo. Committed a correctly-named directory and removed my typo'd one. Render looks nice to me.
